Shared/VideoPlayer.cs

```diff
@@ -65,12 +70,56 @@ public bool IsMuted
 
         public void BufferVideo() => Buffered.Raise();
 
+        public void Seek(TimeSpan timeSpan) => Seeked.Raise(timeSpan);
+
+        public TimeSpan? Duration { get; set; }
+
         public override void Dispose()
         {
             PathChanged?.Dispose();
             base.Dispose();
         }
 
+        public async Task LoadYoutube(string url)
```

Comment:
Devs shouldn't call this. They only set the Path prop and the plugin must handle youtube stuff internally.